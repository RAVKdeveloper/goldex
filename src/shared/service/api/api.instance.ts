import { QueryClient } from '@tanstack/react-query'
import axios from 'axios'

export const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
})

export const api = axios.create({
  baseURL: import.meta.env.VITE_SERVER_URL ?? 'http://23.88.40.65:8000/api/v1',
  headers: {
    'Content-Type': 'application/json',
  },
})
