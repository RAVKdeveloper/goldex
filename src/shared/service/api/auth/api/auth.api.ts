import { api } from '../../api.instance'
import type { LoginDto } from '../dto/login.dto'

class AuthApi {
  async login(userDto: LoginDto) {
    return api.post('/login', userDto)
  }
  async me() {}
}

export const authApi = new AuthApi()
