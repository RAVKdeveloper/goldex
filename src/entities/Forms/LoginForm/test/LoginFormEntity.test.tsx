// import { ReactElement, ReactNode } from 'react'
import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { QueryClientProvider } from '@tanstack/react-query'
import { queryClient } from '@/shared/service/api/api.instance'

import { LoginFormEntity } from '../ui/LoginForm.entity'

const mockedUsedNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}))

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

describe('testing LoginFormEntity component', () => {
  it('render and typing', () => {
    const user = userEvent.setup()

    const component = render(
      <QueryClientProvider client={queryClient}>
        <LoginFormEntity />
      </QueryClientProvider>,
    )

    const rootComponent = screen.getByTestId('loginForm')
    const loginInput = screen.getByTestId('login')
    const passInput = screen.getByTestId('password')

    user.type(loginInput, 'login')
    user.type(passInput, 'password')

    expect(rootComponent).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
})
