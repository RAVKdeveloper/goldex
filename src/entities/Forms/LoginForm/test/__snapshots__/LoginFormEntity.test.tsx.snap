// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing LoginFormEntity component render and typing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
        data-testid="loginForm"
      >
        <div
          class="header"
        >
          <h1
            class="title"
          >
            loginTitle
          </h1>
          <p
            class="supportingText"
          >
            loginSupText
          </p>
        </div>
        <div
          class="content"
        >
          <textfield
            data-testid="login"
            label="login"
            name="login"
            variant="input"
          />
          <textfield
            data-testid="password"
            icon="[object Object]"
            label="password"
            name="password"
            type="password"
            variant="input"
          />
        </div>
        <button
          disabled=""
          variant="primary"
        >
          Войти
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
      data-testid="loginForm"
    >
      <div
        class="header"
      >
        <h1
          class="title"
        >
          loginTitle
        </h1>
        <p
          class="supportingText"
        >
          loginSupText
        </p>
      </div>
      <div
        class="content"
      >
        <textfield
          data-testid="login"
          label="login"
          name="login"
          variant="input"
        />
        <textfield
          data-testid="password"
          icon="[object Object]"
          label="password"
          name="password"
          type="password"
          variant="input"
        />
      </div>
      <button
        disabled=""
        variant="primary"
      >
        Войти
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
